src/compiler/checker.ts

```diff
@@ -13929,6 +13929,17 @@ namespace ts {
                             Diagnostics.The_expected_type_comes_from_the_return_type_of_this_signature,
                         ));
                     }
+                    if ((getFunctionFlags(node) & FunctionFlags.Async) === 0
+                        // exclude cases where source itself is promisy - this way we don't make a suggestion when relating

```

Comment:
Did an existing test case catch this?  If not, would it be worthwhile to add one?

Reply:
Yep. I added the check because there was a `Promise`/`IPromise` test with an arrow where the elaboration seemed bad.